From e33287758f2234d6aabfc5d4e011c4e81e3a47cf Mon Sep 17 00:00:00 2001
From: Hans Wennborg <hans@chromium.org>
Date: Wed, 10 Sep 2025 09:59:13 -0700
Subject: [PATCH] [rust] Remove the old __rust_alloc_error_handler_should_panic
 symbol

Current Rust (rolled in crrev.com/1513010) uses the
__rust_alloc_error_handler_should_panic_v2 function instead, which was
added in crrev.com/1505162.

Bug: 440481922
Change-Id: I2b5c1728748e676907907faffd48e594f4b643c9
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/6935385
Reviewed-by: Arthur Eubanks <aeubanks@google.com>
Commit-Queue: Arthur Eubanks <aeubanks@google.com>
Cr-Commit-Position: refs/heads/main@{#1513787}
---
 build/rust/allocator/lib.rs | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/build/rust/allocator/lib.rs b/build/rust/allocator/lib.rs
index a7fa7a470f0001..b92aec8086f3fc 100644
--- a/build/rust/allocator/lib.rs
+++ b/build/rust/allocator/lib.rs
@@ -96,13 +96,6 @@ mod both_allocators {
         0
     }
 
-    // Mangle the symbol name as rustc expects.
-    // TODO(crbug.com/440481922): Remove this after rolling past https://github.com/rust-lang/rust/pull/143387
-    #[rustc_std_internal_symbol]
-    #[allow(non_upper_case_globals)]
-    #[linkage = "weak"]
-    static __rust_alloc_error_handler_should_panic: u8 = 0;
-
     // Mangle the symbol name as rustc expects.
     #[rustc_std_internal_symbol]
     #[allow(non_upper_case_globals)]
